examples/particle-swarm-optimizer.dx

```diff
@@ -0,0 +1,148 @@
+' # Particle Swarm Optimizer
+
+' ## Fitness function
+
+rosenbrock:: Real -> Real -> Real
+rosenbrock x y = sq (1.0 - x) + 80.0*sq (y - x*x)
+
+
+' We write one that uses vector for input
+
+rosenbrock2:: (2=>Real) -> Real
+rosenbrock2 xs =
+  x = xs.(0@2)
+  y = xs.(1@2)
+  rosenbrock x y
+
+
+' Min should be at 1.0, 1.0
+
+:p rosenbrock 1.0 1.000
+> 0.0
+
+:p rosenbrock2 [1.0, 1.000]
+> 0.0
+
+:p rosenbrock 1.0 1.02
+> 3.2e-2
+
+:p rosenbrock2 [1.0, 1.02]
+> 3.2e-2
+
+
+' ## Helper functions
+
+' ### Rand helpers
+like `ixkey` but takes 2 index to use as keys
+
+ixkey2 :: A n::Ix m::Ix. Key -> n -> m -> Key
+ixkey2 x i = ixkey $ ixkey x i
```

Comment:
This is a potential candidate to move to the prelude.
I started a new demo, and I also want if for that too.

Reply:
It's used in all 3 demo PRs I have open.